### DISCUSSION TITLE
Upload File to Specific Folder using Nuxt

### DISCUSSION THREAD

**Discussion by ldrach** [directus/directus#13555] [Archive]
I have a Nuxt project with a component that allows for file upload. I want to have the files be uploaded to a specific folder other than the root file folder. eg. root/example-target-folder. I cannot seem to find anything in the documentation that details if/how you can upload to a specific folder. I'm implementing the following code from the documentation:

`const form = new FormData(event.target);
await directus.files.createOne(form);`

If anyone has dealt with this before and knows of a solution, it would be much appreciated!

**Comment by rijkvanzanten** [comment]
You can associate more fields of the file by adding more fields to the form data!

See https://docs.directus.io/reference/files/#upload-a-file for an example:

```js
import axios from 'axios';

const fileInput = document.querySelector('input[type="file"]');
const formData = new FormData();

formData.append('title', 'My First File');
formData.append('file', fileInput.files[0]);

await axios.post('/files', formData);
```

**Reply by ldrach** [reply]
To be more specific, this is what I am currently doing:

```js
const formData = new FormData();
formData.append('title', 'title-of-file');
formData.append('file', $event);
```

I have tried some things like appending 'folder' to the formdata which didn't seem to work for getting it into directus in a specific folder either. I do not have a lot of experience with the usage of file uploading. Is there something else I can append to the form data that will tell directus where to store the uploaded file?

**Reply by rijkvanzanten** [reply]
```js
const formData = new FormData();
formData.append('title', 'title-of-file');
formData.append('folder', '8318eca7-bacc-4792-9535-c0329ff7ab94');
formData.append('file', $event);
```

should do the trick! (make sure to replace that fake UUID with the uuid of an actual folder)

**Reply by ldrach** [reply]
Awesome, I knew there was something I was missing! I will give that a go!

**Reply by alsherif-khalaf** [reply]
```
const formData = new FormData(e.target);
            formData.append('title', 'title-of-file');
            formData.append('folder', '52429b6a-1e4c-4d0a-8e0b-0994f545b2c9');
            await directus.files.createOne(formData)
```

It doesn't work ..

the response folder: null

and what if i want to add files in Creating an item :

```
const orders = await directus.items('orders').createOne({
                 patient_age: age,
                 patient_name:  name,
                 patient_gender:  gender,
                 images :  // ??
)}
```

**Reply by ldrach** [reply]
I was able to use the answer @rijkvanzanten gave:

```
const formData = new FormData();
formData.append('title', 'title-of-file');
formData.append('folder', '8318eca7-bacc-4792-9535-c0329ff7ab94');
formData.append('file', $event);
```

And that ended up working for me. In my application, if the user uploads a file, I request the UUID of their folder and use that in the line
`formData.append('folder', '8318eca7-bacc-4792-9535-c0329ff7ab94');`

I was having some trouble with this as well but it turned out I had a mistake with my file upload permissions. After I enabled the proper permissions, it worked great so it might be worth double checking your permissions if it still isn't working for you.

As for your other question, to my understanding, the file upload is separate from the item creation. For context, my project is using a form that allows file uploads. The way I got it working was to upload the file using the code above. When I create the file using:

```
await this.$directus.files.createOne(formData).then(resp => {
        this.file_id = resp.id
      })
```
The ID of the file upload is what needs to be saved with the item being created. So if I am understanding what you are asking, I think you would want to do something like this:

```
const file_id = await this.$directus.files.createOne(formData).then(resp => {
        return resp.id
      })

const orders = await directus.items('orders').createOne({
                 patient_age: age,
                 patient_name:  name,
                 patient_gender:  gender,
            ->   images : file_id 
)}

```

I hope that helps!
